docs/source/operator.rst

```diff
@@ -0,0 +1,149 @@
+.. Copyright (c) 2016, Johan Mabille and Sylvain Corlay
+
+   Distributed under the terms of the BSD 3-Clause License.
+
+   The full license is in the file LICENSE, distributed with this software.
+
+Operators and functions
+=======================
+
+Arithmetic operators
+--------------------
+
+`xtensor` provides overloads of traditional arithmetic operators for ``xexpression`` objects:
+
+- ``operator+(const E&)``
+- ``operator-(const E&)``
+- ``operator+(const E1&, const E2&)``
+- ``operator-(const E1&, const E2&)``
+- ``operator*(const E1&, const E2&)``
+- ``operator/(const E1&, const E2&)``
+
+All these operators are element-wise operators and apply the lazy broadcasting rules explained in
+a previous section.
+
+.. code::
+
+    #incude "xtensor/xarray.hpp"
+
+    xt::xarray<int> a = {{1, 2}, {3, 4}};
+    xt::xarray<int> b = {1, 2};
+
+    xt::xarray<int> res = 2 * (a + b);
+    // => res = {{4, 8}, {8, 12}}
+
+Logical operators
+-----------------
+
+`xtensor` also provides overloads of the logical operators:
+
+- ``operator!(const E&)``
+- ``operator||(const E&)``
+- ``operator&&(const e&)``
+
+Like arithmetic operators, these logical operators are element-wise operators and apply the lazy broadcasting
+rules. In addition to these element-wise logical operators, `xtensor` provides two reducing boolean functions:
+
+- ``any(const E& e)`` returns ``true`` if any of ``e`` elements is truthy, ``false`` otherwise.
+- ``all(const E& e)`` returns ``true`` if all alements of ``e`` are truthy, ``false`` otherwise.
+
+and an element-wise ternary function (similar to the ``: ?`` ternary operator):
+
+- ``where(const E& b, const E1& e&, const E2& e2)`` returns an ``xexpression`` whose elements are those of ``e1``
+  when corresponding elements of ``b`` are thruthy, and those of ``e2`` otherwise.
+
+.. code::
+
+    #include "xtensor/xarray.hpp"
+
+    xt::array<bool> b = { false, true, true, false };
+    xt::array<int> a1 = { 1,   2,  3,  4 };
+    xt::array<int> a2 = { 11, 12, 13, 14 };
+
+    xt::array<int> res = xt::where(b, a1, a2);
+    // => res = { 11, 2, 3, 14 }
+
```

Comment:
maybe add that unlike in `numpy.where`, `xt::where` takes full advantage of the lazyness of xtensor.